Sequentialize to reduce peak memory

dlilien opened the pull request:
Operations from the command line are now mostly sequential. This
should reduce peak memory usage for most operations. Exceptions are
concatenating and loading gecko or osu files.

This should take care of most everything in #17, but let's try to get #16 fixed as well before merging.

dlilien committed:
Sequentialize to reduce peak memory

Operations from the command line are now mostly sequential. This
should reduce peak memory usage for most operations. Exceptions are
concatenating and loading gecko or osu files.

dlilien commented:
@benhills I am hoping permissions are set so that you can push changes to finish #16 and #17 directly to this branch.

dlilien committed:
Make the list copy python 2/3 compatible

dlilien committed:
Fully test processing, with more options to concatenate profiles with differing picks

benhills commented:
Hey, sorry I never got to this. I somehow missed the notifications when they came through. I was about to submit a separate pull request but will deal with this first.

benhills committed:
Move Pulse EKKO partitioning into the load() function

As per issue #16, the PE project files were inconsistently handled
because the partitioning was being done in load_and_exit(). Move that
part of the function into load().

Also, change the loop so that the load() function can handle a mix of
project files and standard PE files.

benhills committed:
Merge branch 'devel' into loading_enhancements

benhills commented:
@dlilien There was a merge conflict above. I took all the accepted the changes from loading_enhancements, but you might want to double check this before we merge the pull request.
